test/e2e/capi_test.go

```diff
@@ -55,18 +55,17 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
 		})
 	})
 
-	// TODO: fix and enable
-	//Context("Running the KCP upgrade spec", func() {
-	//	capi_e2e.KCPUpgradeSpec(context.TODO(), func() capi_e2e.KCPUpgradeSpecInput {
-	//		return capi_e2e.KCPUpgradeSpecInput{
-	//			E2EConfig:             e2eConfig,
-	//			ClusterctlConfigPath:  clusterctlConfigPath,
-	//			BootstrapClusterProxy: bootstrapClusterProxy,
-	//			ArtifactFolder:        artifactFolder,
-	//			SkipCleanup:           skipCleanup,
-	//		}
-	//	})
-	//})
+	Context("Running the KCP upgrade spec", func() {
```

Comment:
are there any other specs that should be enabled before we close #780 ? eg. kcp adoption, experimental features, mhc remediations

Reply:
will look at them again

Reply:
we can also do them in separate PRs if you prefer as long as this PR doesn't close #780 

And if this one only adds the KCP upgrade spec let's make the commit message + title more specific

Reply:
the other tests are not straight forward to run from this repo, they all require specific templates changes that we can't override from the test, so we have to add templates for them. I'm trying to get them working. 
I prefer to do as much as possible in this PR if thats okay with you

Reply:
added two more tests, only the `kcp-adoption` remaining, but having some trouble getting it to work